### PR TITLE
[SPARK-40552][BUILD][INFRA] Upgrade `protobuf-python` to 4.21.6

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Upgrade protobuf-python from 4.21.5 to 4.21.6
[Release notes](https://github.com/protocolbuffers/protobuf/releases/tag/v21.6)

### Why are the changes needed?
[CVE-2022-1941](https://nvd.nist.gov/vuln/detail/CVE-2022-1941)
and
[Github](https://github.com/advisories/GHSA-8gq9-2x98-w8hf)

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by bjornjorgensen** [comment]
cc @grundprinzip

File: dev/requirements.txt [hunk 1]
```diff
@@ -48,4 +48,4 @@ black==22.6.0
 
 # Spark Connect
 grpcio==1.48.1
-protobuf==4.21.5
\ No newline at end of file
+protobuf==4.21.6
```
Comment (HyukjinKwon) [inline dev/requirements.txt:51]: Can you also update `dev/create-release/spark-rm/Dockerfile`?

**Review by HyukjinKwon — APPROVED** [review]
Looks fine. cc @grundprinzip FYI

**Review by itholic — APPROVED** [review]
Looks good

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**